test/verify.test.js

```diff
@@ -307,77 +309,129 @@ test.serial('Verify "assets" is an Array of Object with a glob Arrays in path pr
   t.true(gitlab.isDone());
 });
 
-test('Throw SemanticReleaseError if "assets" option is not a String or an Array of Objects', async t => {
+test.serial('Throw SemanticReleaseError if "assets" option is not a String or an Array of Objects', async t => {
+  const owner = 'test_user';
+  const repo = 'test_repo';
   const env = {GITLAB_TOKEN: 'gitlab_token'};
   const assets = 42;
+  const gitlab = authenticate(env)
+    .get(`/projects/${owner}%2F${repo}`)
+    .reply(200, {permissions: {project_access: {access_level: 40}}});
 
-  const [error] = await t.throwsAsync(
+  const aggregateError = await t.throwsAsync(
     verify(
       {assets},
-      {env, options: {repositoryUrl: 'https://gitlab.com/semantic-release/gitlab.git'}, logger: t.context.logger}
-    )
+      {env, options: {repositoryUrl: `https://gitlab.com/${owner}/${repo}.git`}, logger: t.context.logger}
+    ),
+    AggregateError,
+    'Error thrown but of unexpected type.'
   );
+  const errors = [...aggregateError];
 
-  t.is(error.name, 'SemanticReleaseError');
-  t.is(error.code, 'EINVALIDASSETS');
+  const expectedNumOfErrors = 1;
```

Comment:
What's the point of adding the variable `expectedNumOfErrors`? It's only used on the next line and it's different for each test. I'd rather just pass the number directly to the test.

I prefer to have test that are simple to read and understand. Adding a bunch intermediate variable just create more code, more possibilities for a mistake and more cognitive load to read the tests.

Reply:
Actually I added the intermediate to make it more readable and easier to understand, and to keep the DRY principle. (It is used in the assert and in the assert error message => you don't have to replace it at two places => less bugs). Even though it is one more constant it is more expressive : it is what it says it is : the expected number of errors. Without it you would have to guess it. So I am using this approach to actually lower the cognitive load.

Reply:
Actually I added the intermediate to make it more readable and easier to understand, and to keep the DRY principle. (It is used in the assert and in the assert error message => you don't have to replace it at two places => less bugs). Even though it is one more constant it is more expressive : it is what it says it is : the expected number of errors. Without it you would have to guess it. So I am using this approach to actually lower the cognitive load.

Reply:
The added message is not really useful. It just state the obvious. It's number comparison and `ava` out the expect line and the value it was passed too. Adding a sentence to state that number `x` is different than number `y` doesn't aadd any value vs `ava` output other than having to write more code.

Reply:
Just to explain my point (and after you consider it, since you are the maintainer, I will follow whatever rules you see fit):

Here is the `ava`'s output without the failure message (I put the expected to `2` just to make it fail):
![image](https://user-images.githubusercontent.com/1336689/67514227-a9772680-f69c-11e9-8ccc-14b8db43f365.png)

![image](https://user-images.githubusercontent.com/1336689/67514297-c875b880-f69c-11e9-83f7-d3ba4d659a20.png)

And here is the same with the failure message:
![image](https://user-images.githubusercontent.com/1336689/67514567-3ae69880-f69d-11e9-86b4-08efee43b890.png)

![image](https://user-images.githubusercontent.com/1336689/67514477-14c0f880-f69d-11e9-98a8-cabcf90f40d9.png)

Without the message, you have no clue of what is going and have to go to the detailed failure report, to read the code, to try to understand what it means to finally figure out what the problem is about.

With the message,  you immediately have an explanation next to the test in the initial summary, and for every test failing. No need to dig in: you have an overview in the summary of why each test has failed. You then can go to the detailed failure report if need be to get more insight.

Now, the message itself night not have been the smartest, and could be changed to something like  "Only one error should have been thrown but more have." And this way, no need for an `expectedNumOfErrors`. But I do see an advantage in having a message.

What do you think?